### PR TITLE
Correctly handle presence of multiple plans with the same name

### LINKED ISSUE #1292: ReferencesNonexistentServicePlan creating instance using externalServicePlanName
I'm seeing an error provisioning a instance using canary images and `externalServicePlanName`.

The error message looks wrong. There is a plan `default` for this service class. It looks like the problem is that more than there is more than one plan with external name `default`. In fact, this broker creates a plan for each service class all called `default`.

```
I0928 20:06:13.214750      24 controller_instance.go:509] Adding/Updating ServiceInstance test/mariadb-persistent-rcc88
--
  | I0928 20:06:13.214760      24 controller.go:421] looking up a ServiceClass from Ref: "a4fb4cda-a471-11e7-bf4a-2e266f501f05"
  | I0928 20:06:13.220961      24 controller.go:398] Found 12 ServicePlans based on externalName "default"
  | W0928 20:06:13.220995      24 controller.go:254] ServiceInstance "test/mariadb-persistent-rcc88" references a non-existent ServicePlan "default" on ServiceClass "mariadb-persistent"
  | I0928 20:06:13.221014      24 controller_instance.go:1122] Setting ServiceInstance "test/mariadb-persistent-rcc88" condition "Ready" to False
  | I0928 20:06:13.221022      24 controller_instance.go:1188] Updating Ready condition for ServiceInstance test/mariadb-persistent-rcc88 to False
  | I0928 20:06:13.224451      24 controller.go:218] Error syncing ServiceInstance test/mariadb-persistent-rcc88: ServiceInstance "test/mariadb-persistent-rcc88" references a non-existent ServicePlan "default" on ServiceClass "mariadb-persistent"
```

Here is the YAML of my instance:

```yaml
apiVersion: servicecatalog.k8s.io/v1alpha1
kind: ServiceInstance
metadata:
  creationTimestamp: '2017-09-28T17:58:45Z'
  finalizers:
    - kubernetes-incubator/service-catalog
  generateName: mariadb-persistent-
  generation: 1
  name: mariadb-persistent-rcc88
  namespace: test
  resourceVersion: '60'
  selfLink: >-
    /apis/servicecatalog.k8s.io/v1alpha1/namespaces/test/serviceinstances/mariadb-persistent-rcc88
  uid: abb57564-a476-11e7-b83f-0242ac110006
spec:
  externalID: c191cc77-6c83-421c-9ca6-a52962d9d5d1
  externalServiceClassName: mariadb-persistent
  externalServicePlanName: default
  parametersFrom:
    - secretKeyRef:
        key: parameters
        name: mariadb-persistent-parameters-h6flh
  userInfo:
    extra:
      scopes.authorization.openshift.io:
        - 'user:full'
    groups:
      - 'system:authenticated:oauth'
      - 'system:authenticated'
    uid: ''
    username: sam
status:
  asyncOpInProgress: false
  conditions:
    - lastTransitionTime: '2017-09-28T17:58:45Z'
      message: >-
        The instance references a ServicePlan that does not exist.
        ServiceInstance "test/mariadb-persistent-rcc88" references a
        non-existent ServicePlan "default" on ServiceClass "mariadb-persistent"
      reason: ReferencesNonexistentServicePlan
      status: 'False'
      type: Ready
  reconciledGeneration: 0
```

@pmorie @vaikas-google

### PR DESCRIPTION
**Pull request by pmorie**
...when looking up the service plan for an instance.

Fixes #1292

### CODE REVIEW

File: pkg/registry/servicecatalog/serviceplan/storage.go [hunk 1]
```diff
@@ -90,6 +90,7 @@ func Match(label labels.Selector, field fields.Selector) storage.SelectionPredic
 // toSelectableFields returns a field set that represents the object for matching purposes.
 func toSelectableFields(servicePlan *servicecatalog.ServicePlan) fields.Set {
 	spSpecificFieldsSet := make(fields.Set, 2)
```
Comment (vaikas-google) [inline pkg/registry/servicecatalog/serviceplan/storage.go]: Change 2 to 3 since you add a field.

**Comment by MHBauer** [comment]
Test case to make sure this fails appropriately?

**Comment by spadgett** [comment]
I can confirm this fixes the error I was seeing and the service provisions successfully.

`serviceClassRef` and `servicePlanRef` are still not being set on the instance, but @pmorie mentioned that's another known problem.

**Comment by vaikas-google** [comment]
Could we hold off on this I'll add these to my changes (wrt. setting Service[Plan|Class]Ref). PR coming very shortly.

**Comment by pmorie** [comment]
This has been incorporated into #1305

**Pull request closed without merging**